File: 0206-reverse-linked-list/0206-reverse-linked-list.dart
/**
 * Definition for singly-linked list.
 * class ListNode {
 *   int val;
 *   ListNode? next;
 *   ListNode([this.val = 0, this.next]);
 * }
 */
class Solution {
  ListNode? reverseList(ListNode? head) {
    if (head == null) {
        return head;
    }
    ListNode answer = ListNode(head.val);
    while (head?.next != null) {
        head = head?.next;
        answer = ListNode(head!.val, answer);
    }

    return answer;
  }
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return head
        answer = ListNode(head.val)
        while head.next:
            head = head.next
            answer = ListNode(head.val, answer)
        return answer